base/collection.go

```diff
@@ -116,8 +116,17 @@ func GetCollectionFromCluster(cluster *gocb.Cluster, spec BucketSpec, waitUntilR
 	// Safe to get first node as there will always be at least one node in the list and cluster compat is uniform across all nodes.
 	clusterCompatMajor, clusterCompatMinor := decodeClusterVersion(nodesMetadata[0].ClusterCompatibility)
 
+	c := bucket.DefaultCollection()
```

Comment:
I don't think bucket.DefaultCollection() is particularly lightweight - we'd be better off only allocating when we're actually using the default collection.
Given that bucket.DefaultCollection is just a helper function for bucket.Scope("_default").Collection("_default), maybe just write a helper function for bucketSpec to return scope and collection name that will do the appropriate substitution of default values?